services/venice-server/src/test/java/com/linkedin/venice/listener/StorageReadRequestHandlerTest.java

```diff
@@ -192,6 +217,27 @@ public void cleanUp() {
         readMetadataRetriever,
         serverConfig,
         context);
+    ReadResponseLatencyInjector.removeExtraLatency();
+  }
+
+  private enum ParallelQueryProcessing {
+    PARALLEL(true), SEQUENTIAL(false);
+
+    final boolean configValue;
+
+    ParallelQueryProcessing(boolean configValue) {
+      this.configValue = configValue;
+    }
+  }
+
+  @DataProvider(name = "storageReadRequestHandlerParams")
+  public Object[][] storageReadRequestHandlerParams() {
+    return new Object[][] { { ParallelQueryProcessing.SEQUENTIAL, 10, ValueSize.SMALL_VALUE },
```

Comment:
For symmetry's sake, do we need 

ParallelQueryProcessing.SEQUENTIAL, 100, ValueSize.LARGE_VALUE
ParallelQueryProcessing.SEQUENTIAL, 100, ValueSize.SMALL_VALUE

Cases?

Reply:
I don't think we "need" it, but why not :D ... I tweaked the params so that this is a recordCount param rather than chunkSize param to make it less error-prone, and added a couple extra permutations.